src/spuco/last_layer_retrain/dispel.py

```diff
@@ -301,5 +306,10 @@ def train(self):
             self.class_weight_options = [{c: 1 for c in range(n_class)}]
 
         self.hyperparam_selection(X_labeled_train, y_labeled_train, g_labeled_train, X_labeled_val, y_labeled_val, g_labeled_val)
-        coef, intercept = self.train_multiple_model(self.best_alpha, self.best_s, self.best_C, X_labeled_train, y_labeled_train, g_labeled_train, self.best_class_weight)
+        if self.validation_set is not None:
+            X_labeled = np.concatenate((X_labeled, X_labeled_val))
+            y_labeled = np.concatenate((y_labeled, y_labeled_val))
+            g_labeled = np.concatenate((g_labeled, g_labeled_val))
```

Comment:
if validation set is provided, both the labeled and unlabaled set can be used for trainiing, so put them together